odoo/models.py

```diff
@@ -1367,6 +1367,10 @@ def load_views(self, views, options=None):
         :param options['action_id']: id of the action to get the filters
         :return: dictionary with fields_views, fields and optionally filters
         """
+        # populate the context with essential information for debugging problematic views
+        # all the way up the stack
+        self = self.with_context(loaded_view=(self._name, views, options))
```

Comment:
Why don't you simply pass `self`?
```suggestion
        self = self.with_context(load_views_args=(self, views, options))
```